src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/NodeStatsAllShardsMetricsCollectorTests.java

```diff
@@ -49,9 +52,32 @@ public void tearDown() throws Exception {
         super.tearDown();
     }
 
+    @Test
+    public void testGetNodeIndicesStatsByShardField() {
+        try {
+            nodeStatsAllShardsMetricsCollector.getNodeIndicesStatsByShardField();
+        } catch (Exception e) {
+            assertTrue("There shouldn't be any exception in the code; Please check the reflection code for any changes", true);
+        }
+    }
+
+    @Test
+    public void testGetMetricsPath() {
+        String expectedPath = PluginSettings.instance().getMetricsLocation()
+            + PerformanceAnalyzerMetrics.getTimeInterval(startTimeInMills)+ "/" + PerformanceAnalyzerMetrics.sIndicesPath + "/NodesStatsIndex/55";
+        String actualPath = nodeStatsAllShardsMetricsCollector.getMetricsPath(startTimeInMills, "NodesStatsIndex", "55");
+        assertEquals(expectedPath, actualPath);
+
+        try {
+            nodeStatsAllShardsMetricsCollector.getMetricsPath(startTimeInMills, "NodesStatsIndex");
+            assertTrue("Negative scenario test: Should have been a RuntimeException", true);
+        } catch (RuntimeException ex) {
+            //- expecting exception...only 1 values passed; 2 expected
```

Comment:
if we are expecting exception here, should we return here and leave a Assert.fail() at the end of the try block ?

Reply:
good catch, thanks.